src/backends/torch/torchlib.cc

```diff
@@ -534,17 +591,25 @@ namespace dd
     try
       {
         inputc.transform(ad);
+
+        _module._native = std::shared_ptr<NativeModule>(
+            NativeFactory::from_template(_template, _template_params, inputc));
```

Comment:
In Caffe backend, we try to instantiate the network template at service init. Do we want to do the same here ? As a side question, probably the template could be instantiated before the input are even transformed ?

Reply:
unfortunately, instanciating nbeats needs the forecast and backcast length to be known, and they are "timesteps" input connector parameter, which can be given only at train/predict time

Reply:
this is because for lstm, timestep is only used at timeserie length and lstm simply unroll over this data dimension, but for nbeats this has to be statically fixed

Reply:
OK, maybe we can do it within `graph` ?